esphome/components/da217/sensor.py

```diff
@@ -0,0 +1,295 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_ID,
+    ICON_BRIEFCASE_DOWNLOAD,
+    CONF_RESOLUTION,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_METER_PER_SECOND_SQUARED,
+)
+
+DEPENDENCIES = ["i2c"]
+
+da217_ns = cg.esphome_ns.namespace("da217")
+
+CONF_ACCEL_X = "accel_x"
+CONF_ACCEL_Y = "accel_y"
+CONF_ACCEL_Z = "accel_z"
+
+# RESOLUTION_RANGE
+
+CONF_ENABLE_HIGH_PASS_FILTER = "enable_high_pass_filter"
+CONF_ENABLE_WATCHDOG = "enable_watchdog"
+CONF_WATCHDOG_TIME = "watchdog_time"
+CONF_FULL_SCALE = "full_scale"
+
+WatchdogTime = da217_ns.enum("WatchdogTime")
+WATCHDOG_TIMES = {
+    "1MS": WatchdogTime.TIME1MS,
+    "50MS": WatchdogTime.TIME50MS,
+}
```

Comment:
Please make all of the enums readable with the better/descriptive naming.
```suggestion
WATCHDOG_TIMES = {
    1: WatchdogTime.WATCHDOG_TIME_1_MS,
    50: WatchdogTime.WATCHDOG_TIME_50_MS,
}
```